plugins/metrics/pkg/backend/metrics.go

```diff
@@ -107,6 +109,45 @@ func (m *MetricsBackend) requestNodeSync(ctx context.Context, cluster *corev1.Re
 	).Info("node sync requested")
 }
 
+func (m *MetricsBackend) onSyncUpdateClusterResources(ctx context.Context, cfg *node.MetricsCapabilityConfig) {
+	shouldDeployCollector := false
+	defer func() {
+		m.Logger.Debugf("gateway metrics otel collector should exist : %v", shouldDeployCollector)
+		collectorAddress, err := m.ClusterDriver.ConfigureOTELCollector(shouldDeployCollector)
+		if err != nil {
+			m.Logger.Errorf("failed to apply collector updates: %v", err)
+		}
+		m.OTELForwarder.SetAddress(collectorAddress)
+	}()
+	// TODO : #1176 should check for otel capability
+	if cfg.GetSpec() != nil && cfg.GetEnabled() {
+		shouldDeployCollector = true
+		return
+	}
+	cls, err := m.MgmtClient.ListClusters(ctx, &managementv1.ListClustersRequest{})
+	if err != nil {
+		return
+	}
+	for _, cl := range cls.Items {
+		if cl.Metadata == nil {
+			continue
+		}
+		for _, cap := range cl.Metadata.Capabilities {
+			if cap.Name == wellknown.CapabilityMetrics {
+				spec, err := m.GetNodeConfiguration(ctx, cl.Reference())
+				if err != nil {
+					m.Logger.Warnf("failed to get node configuration for cluster %s: %v", cl.Reference().GetId(), err)
+					continue
+				}
+				if spec != nil { // TODO : #1176 check for otel capability
+					shouldDeployCollector = true
```

Comment:
Do we actually do anything with this value?

Reply:
it's captured in the defer func in order to tell the cluster driver how to deploy otel gateway resources
https://github.com/rancher/opni/blob/c8469d078f588c0cbc4ee769ce1bd756b6e5b87e/plugins/metrics/pkg/backend/metrics.go#L114-L120

Reply:
I think this function is hard to read though, so let me rework this a little